config/config-api/src/com/thoughtworks/go/util/command/EnvironmentVariableContext.java

```diff
@@ -227,25 +224,21 @@ public int hashCode() {
         return properties != null ? properties.hashCode() : 0;
     }
 
-    public String reportText() {
-        throw new RuntimeException("IMPLEMENT ME");
-    }
-
-    public void setupRuntimeEnvironment(Map<String, String> env, ConsoleOutputStreamConsumer consumer) {
+    public List<String> report() {
```

Comment:
There's a bit of a behavior change here. The old `env` map was pre-filled. This new `names` set is not. Earlier call to `setupRuntimeEnvironment` was passing in the existing environment (which would be a combination of environment-level, pipeline-level and stage-level variables and these could be overridden by a job. With this change, I suspect that is broken. Not completely sure.